lib/codec.js

```diff
@@ -0,0 +1,165 @@
+/*!
+ * Copyright (c) 2020 Digital Bazaar, Inc. All rights reserved.
+ */
+import {termCodecs} from './codecs/codecMap';
+import {Base64PadCodec} from './codecs/base64Pad';
+import {CborldError} from './error';
+import {CodecMapCodec} from './codecs/codecMap';
+import {UrlCodec} from './codecs/url';
+import {ContextCodec} from './codecs/context';
+import {SimpleTypeCodec} from './codecs/simpleType';
+import {XsdDateTimeCodec} from './codecs/xsdDateTime';
+
+/**
+ * Gets a list of all term codecs available to use when transforming term
+ * values.
+ *
+ * @returns {Array} - An array of all term codecs.
+ */
+export function getTermCodecs() {
+  const codecs = [];
+  termCodecs.forEach((value, key) => {
+    if(typeof key === 'string') {
+      codecs.push(key);
+    }
+  });
+
+  return codecs;
+}
+
+/**
+ * Gets a compression map for the given set of JSON-LD Context URLs.
+ *
+ * @param {object} [args] - The arguments to the function.
+ * @param {Array} [args.contextUrls] - The JSON-LD Context URLs to use when
+ *   building the compression map.
+ * @param {object} [args.appTermMap] - A map of JSON-LD terms and
+ *   their associated CBOR-LD compression codecs.
+ * @param {object} [args.documentLoader] - The document loader to use when
+ *   dereferencing URLs.
+ * @param {object} [args.encode] - If true, a map of term encoding codecs are
+ *   returned.
+ * @param {object} [args.decode] - If true, a map of term decoding codecs are
+ *   returned.
+ *
+ * @returns {object} - The compression map.
+ */
+export async function getTermCodecMap({
```

Comment:
Docs refer to this as a "compression map". Which is the better name? It's used for both compression and decompression, I presume. If we stick with "term codec map" can we mention that a bit more in the docs?

Reply:
I ended up settling on codec map... because we use the map for both encoding and decoding.

Reply:
Fixed in 9334d8e420e99c10ec5858eff5b59d4f7e13c071.